src/Orchard.Web/Modules/Orchard.Autoroute/Handlers/AutoroutePartHandler.cs

```diff
@@ -54,6 +58,15 @@ public class AutoroutePartHandler : ContentHandler {
             ProcessAlias(part);
         }
 
+        private string LockString {
+            get {
+                return string.Join(".",
+                    _orchardServices.WorkContext?.CurrentSite?.BaseUrl ?? "",
+                    _orchardServices.WorkContext?.CurrentSite?.SiteName ?? "",
+                    "AutoroutePartHandler");
```

Comment:
Locking per Url and SiteName ... good idea 👍 
I started looking at async await methods but this does the trick 😄